### PR TITLE
[Feed] Save to ref manager on feed card 

### PR DESCRIPTION
**Pull request by yattias**
- Save to ref manager directly in feed card
- Rendering citations, peer review score and others on feed card

Screenshot 2024-04-10 at 2 16 37 PM

### CODE REVIEW

File: components/Author/Tabs/FeedCard.tsx
Comment (yattias) [inline components/Author/Tabs/FeedCard.tsx:1]: Just as an FYI, this component might be the worst frontend component we have in our codebase and needs to be fully rewritten. Until we do so, the strategy is to add limited code to it and instead, add child components

File: components/Author/Tabs/FeedCard.tsx [hunk 1]
```diff
@@ -285,6 +273,17 @@ function FeedCard({
   const bountyAmount = documentFilter?.bounty_total_amount;
   const hasActiveBounty = documentFilter?.bounty_open;
 
+  let numOfVisibleHubs = 3;
```
Comment (yattias) [inline components/Author/Tabs/FeedCard.tsx:276]: Ideally this display logic would be done via CSS but given how the component is structured, it is not possible to do so without rewriting or relying on JS.

File: components/Author/Tabs/FeedCard.tsx [hunk 2]
```diff
@@ -434,59 +465,64 @@ function FeedCard({
...
-                      <div className={css(styles.desktop)}>
+
+                      <div className={css(styles.desktop, styles.badges)}>
                         <DocumentHubs
                           hubs={parsedHubs}
                           withShowMore={false}
                           hideOnSmallerResolution={true}
+                          numOfVisibleHubs={numOfVisibleHubs}
                         />
+                        {hasActiveBounty && (
```
Comment (yattias) [inline components/Author/Tabs/FeedCard.tsx:476]: This is not new code, moved around

File: components/Feed/FeedCardActivity.tsx
Comment (yattias) [inline components/Feed/FeedCardActivity.tsx:1]: This new component renders the feed card activity state (comments, citations, etc..)

File: components/Search/SearchResultsForDocs.js [hunk 3]
```diff
@@ -582,8 +582,9 @@ const SearchResultsForDocs = ({ apiResponse, entityType, context }) => {
             return (
               <FeedCard
                 {...post}
+                unifiedDocumentId={post.unified_document_id}
                 formattedDocType={"post"}
-                key={post?.id || index}
```
Comment (koutst) [inline components/Search/SearchResultsForDocs.js:586]: Is this index fallback not necessary anymore?

**Review by koutst — APPROVED** [review]
LGTM

File: components/Search/SearchResultsForDocs.js (hunk 3 above)
Comment (yattias) [reply components/Search/SearchResultsForDocs.js:586]: Yeah, it wasn't necessary to begin with

**Pull request merged**